src/erc721/ERC721.sol

```diff
@@ -67,22 +66,45 @@ contract ERC721 {
         );
     }
     
+    /*///////////////////////////////////////////////////////////////
+                              ERC-20-LIKE LOGIC
+    //////////////////////////////////////////////////////////////*/
+    
+    function transfer(address to, uint256 tokenId) external payable {
+        require(msg.sender == ownerOf[tokenId], "NOT_OWNER");
+        
+        // This is safe because ownership is checked
+        // against decrement, and sum of all user
+        // balances can't exceed type(uint256).max!
+        unchecked {
+            balanceOf[msg.sender]--; 
+        
+            balanceOf[to]++;
+        }
+        
+        delete getApproved[tokenId];
+        
+        ownerOf[tokenId] = to;
+        
+        emit Transfer(msg.sender, to, tokenId); 
+    }
+    
     /*///////////////////////////////////////////////////////////////
                               ERC-721 LOGIC
     //////////////////////////////////////////////////////////////*/
     
-    function supportsInterface(bytes4 sig) external pure returns (bool) {
-        return (sig == 0x80ac58cd || sig == 0x5b5e139f);
+    function supportsInterface(bytes4 interfaceId) external pure returns (bool supported) {
+        supported = interfaceId == 0x80ac58cd || interfaceId == 0x5b5e139f;
     }
     
-    function approve(address spender, uint256 tokenId) external {
+    function approve(address spender, uint256 tokenId) external payable {
```

Comment:
why payable?

Reply:
tbh, I think `payable` was an odd choice for EIP authors. I believe they wanted to support royalty payments, and see 'deeds' or digital art as more likely to take advantage of 'fee-on-transfer' functions. Can punt this into inheritance, but just added to flag. 
![image](https://user-images.githubusercontent.com/41117279/132999736-bca57809-9da2-400f-ad9e-446292b2bfe8.png)

It is also especially odd to include payability on approvals..... as the fee should be grabbed on `transferFrom`.....